test/e2e/specs/editor/plugins/meta-attribute-block.spec.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+const VARIATIONS = [
+	[ 'Early Registration', 'test/test-meta-attribute-block-early' ],
+	[ 'Late Registration', 'test/test-meta-attribute-block-late' ],
+];
+
+test.describe( 'Block with a meta attribute', () => {
+	test.beforeAll( async ( { requestUtils } ) => {
+		await requestUtils.activatePlugin(
+			'gutenberg-test-meta-attribute-block'
+		);
+	} );
+
+	test.afterAll( async ( { requestUtils } ) => {
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-meta-attribute-block'
+		);
+	} );
+
+	for ( const [ title, blockName ] of VARIATIONS ) {
+		test.describe( title, () => {
+			test( 'Should persist the meta attribute properly', async ( {
+				admin,
+				editor,
+				page,
+				pageUtils,
+			} ) => {
+				await admin.createNewPost();
+				await editor.insertBlock( { name: blockName } );
+				await page.keyboard.type( 'Value' );
```

Comment:
I'm wondering if typing immediately after inserting a block can potentially be flaky since we don't have a promise ensuring the block is there and focused (AFAIR there were issues with that before!). What if we returned the inserted block locator from the `createBlock` so that we can do a simple `await expect( block ).toBeFocused()`? I've drafted an implementation in  https://github.com/WordPress/gutenberg/pull/55851.

Reply:
Would look more less like this:

```js
const paragraph = await editor.insertBlock( { name: 'core/paragraph' } );
await expect( paragraph ).toBeFocused();
await page.keyboard.type( 'Howdy hoo!');
```

Reply:
This should be okay. The editor will automatically focus first focusable element in newly inserted block.

Reply:
Is the block insertion synchronous, though?

Reply:
Right, maybe we should await that. @kevin940726, WDYY?

I'll keep an eye on the flaky reports for this test, just in case.